### PR TITLE
Blob copy routines

### PR DESCRIPTION
**Pull request by agr**
Porting blob copy routines from https://github.com/NuGet/NuGet.Jobs/blob/main/src/Catalog/Persistence/
Added the ability to request blob metadata when listing blobs and a function to update it.
Moved some particularly noisy logs under `Verbose` flag.

### CODE REVIEW

File: src/NuGet.Services.Storage/AggregateStorage.cs [hunk 1]
```diff
@@ -33,7 +33,18 @@ public class AggregateStorage : Storage
...
+        protected override Task OnCopyAsync(
+            Uri sourceUri,
+            IStorage destinationStorage,
+            Uri destinationUri,
+            IReadOnlyDictionary<string, string> destinationProperties,
+            CancellationToken cancellationToken)
+        {
+            throw new NotImplementedException();
+        }
+
```
Comment (joelverhagen) [inline src/NuGet.Services.Storage/AggregateStorage.cs:46]:
```suggestion
```

**Review by joelverhagen — APPROVED** [review]

**Comment by zhhyu** [comment]
Looking ...

File: src/NuGet.Services.Storage/AzureStorage.cs [hunk 2]
```diff
@@ -11,26 +11,47 @@
...
+
+        public AzureStorage(
+            CloudStorageAccount account,
+            string containerName,
+            string path,
+            Uri baseAddress,
+            bool useServerSideCopy,
+            bool initializeContainer,
+            ILogger<AzureStorage> logger)
+
```
Comment (zhhyu) [inline src/NuGet.Services.Storage/AzureStorage.cs:32]: nit: empty line.

File: src/NuGet.Services.Storage/Storage.cs [hunk 3]
```diff
@@ -28,10 +29,41 @@ public override string ToString()
...
+            {
+                await OnCopyAsync(sourceUri, destinationStorage, destinationUri, destinationProperties, cancellationToken);
+            }
+            catch (Exception e)
+            {
+                TraceException(nameof(CopyAsync), sourceUri, e);
+                throw;
+            }
+
+            TraceExecutionTime(nameof(CopyAsync), sourceUri, stopwatch.ElapsedMilliseconds);
```
Comment (zhhyu) [inline src/NuGet.Services.Storage/Storage.cs:63]: Should we stop the stopwatch, or it doesn't matter?

Comment (zhhyu) [reply src/NuGet.Services.Storage/Storage.cs:63]: Never mind, it doesn't need: https://stackoverflow.com/questions/24140261/should-i-stop-stopwatch-at-the-end-of-the-method

**Review by zhhyu — APPROVED** [review]

**Pull request merged**